### PR TITLE
Allow consecutive runs of subctl join

### PR DESCRIPTION
**Pull request by mangelajo**
This will make sure that we update the Submariner operator resource in the
server

Fixes issue: #49

---
This change is [Reviewable](https://reviewable.io/reviews/submariner-io/submariner-operator/79)

### CODE REVIEW

**Comment by mangelajo** [comment]
@skitt sorry, I just realized this was assigned to you, ...

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -258,7 +258,7 @@ for i in 2 3; do
                         --nattport ${ce_ipsec_nattport} \
                         --ikeport ${ce_ipsec_ikeport} \
                         --colorcodes ${subm_colorcodes} \
-			--disable-nat \
+			                  --disable-nat \
```
Comment (skitt) [inline scripts/kind-e2e/e2e.sh]: If you’re going to fix this, please use spaces throughout; there are still tabs at the start of the line ;-).

File: scripts/kind-e2e/e2e.sh [hunk 2]
```diff
@@ -311,6 +311,18 @@ for i in 2 3; do
...
+
+../bin/subctl join --operator-image submariner-operator:local \
+                --kubeconfig ${PRJ_ROOT}/output/kind-config/dapper/kind-config-$context \
+                --clusterid ${context} \
+                --repository ${subm_engine_image_repo} \
+                --version ${subm_engine_image_tag} \
+                --nattport ${ce_ipsec_nattport} \
+                --ikeport ${ce_ipsec_ikeport} \
+                --colorcodes ${subm_colorcodes} \
+                --disable-nat broker-info.subm
```
Comment (skitt) [inline scripts/kind-e2e/e2e.sh]: Excellent, I was going to ask for that!

File: pkg/subctl/operator/deploy/ensure.go [hunk 3]
```diff
@@ -42,24 +43,29 @@ func Ensure(config *rest.Config, submarinerSpec submariner.SubmarinerSpec) error
...
-		}
-	}
 
-	// TODO follow ensure pattern:
-	// if created, err := crs.Ensure(...); err != nil {
-	//	return err
-	// } else if created {
-	//	fmt.Printf("* Created Submariner CR.\n")
-	// }
+	updateOrCreateSubmariner(submarinerClient, namespace, submariner)
```
Comment (skitt) [inline pkg/subctl/operator/deploy/ensure.go]: This needs to be checked for errors.

Comment (mangelajo) [reply pkg/subctl/operator/deploy/ensure.go]: oh true, thanks

File: scripts/kind-e2e/e2e.sh (hunk 1 above)
Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh]: ack, thanks

**Comment by mangelajo** [comment]
@skitt , I addressed your comments, and also converted the comment about second time run for subctl into an echo because it was weird to see subctl being run again for "no specific reason".

**Comment by mangelajo** [comment]
@skitt have an eye, comments addressed :-)

**Comment by dfarrell07** [comment]
> It failed the e2e tests for me, but I see it passing in CI so I will try again/debug in the morning. Looks good statically, non-blocker.

Re-running over the night, all passed/looks good in the morning. Thanks!

File: pkg/subctl/operator/deploy/ensure.go [hunk 4]
```diff
@@ -58,24 +59,33 @@ func Ensure(config *rest.Config, submarinerSpec submariner.SubmarinerSpec) error
...
+	_, err := clientSet.SubmarinerV1alpha1().Submariners(namespace).Create(submariner)
+	if err == nil {
+		return true, nil
+	} else if errors.IsAlreadyExists(err) {
+		existingCfg, err := clientSet.SubmarinerV1alpha1().Submariners(namespace).Get(submariner.Name, v1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to get pre-existing cfg %s : %s", submariner.Name, err)
+		}
+		submariner.ResourceVersion = existingCfg.ResourceVersion
+		_, err = clientSet.SubmarinerV1alpha1().Submariners(namespace).Update(submariner)
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply pkg/subctl/operator/deploy/ensure.go]: Oh, I didn't know we had those helpers ("CreateOrUpdate") I think we should switch to that lots of our code, thanks for the suggestion @tpantelis  I opened this bug: https://github.com/submariner-io/submariner-operator/issues/90

**Pull request merged**